frontend/public/script.js

```diff
@@ -129,25 +129,86 @@ if (createRecord) {
   closeFunc(closeSale, createRecordDiv);
 }
 
-const apiUrl = "http://localhost:4000/api/v1/user";
+const apiUrl = "http://localhost:4000/api/v1/users/currentUser"; // replace with "your_url//api/v1/users/currentUser"
+const dbUrl = "http://localhost:4000/api/v1/users/userDb"; // replace with "your_url//api/v1/users/userDb"
 
 // fetch data from api here
+
+const updateNavbar = (username) => {
+  const usernames = document.querySelectorAll("#username");
+  usernames.forEach((user) => {
+    user.innerHTML = username || "John Doe";
+  });
+};
+
+const updateWelcomeMsg = (user) => {
+  const welcomeMsg = document.querySelectorAll("#welcome-msg");
+  welcomeMsg.forEach((msg) => {
+    msg.innerHTML = `Welcome ${user.username}!`;
+  });
+};
+
 const fetchCurrentUser = async () => {
   try {
     const response = await fetch(apiUrl);
     const data = await response.json();
     updateNavbar(data.username);
+    updateWelcomeMsg(data);
   } catch (error) {
     console.error("Error fetching user:", error);
   }
 };
 
-const updateNavbar = (username) => {
-  const usernames = document.querySelectorAll("#username");
-  usernames.forEach((user) => {
-    user.innerHTML = username || "John Doe";
+// function to dynamically set the href of each attendant to his/her sales page from the navbar
+const setHref = (value) => {
+  const users = document.querySelectorAll(".username");
+  users.forEach((user) => {
+    user.href = value;
+  });
+};
+
+// function to dynamically set the href of each attendant to his/her sales page from the users page
+const setUserHref = (database) => {
+  const userLink = document.querySelectorAll(".user-link");
+
+  userLink.forEach((user) => {
+    const uiUsername = user.innerHTML;
+    database.forEach((data) => {
+      if (uiUsername === data.username) {
+        if (data.role === "Admin") {
+          user.href = "#";
+        }
+        if (data.role === "Attendant") {
+          user.href = `/api/v1/users/sales/${data.id}`;
+        }
+      }
+    });
   });
 };
 
+// parent function to dynamically set the href of each attendant to his/her sales page
+const updateUserLink = async () => {
+  try {
+    const response = await fetch(apiUrl);
+    const databaseRes = await fetch(dbUrl);
```

Comment:
I am kind of confused here...What is the essence of making a call to the database? Can we have a call to go through this PR together?